crates/bevy_ecs/src/reflect/component.rs

```diff
@@ -273,12 +276,22 @@ impl<C: Component + Reflect + TypePath> FromType<C> for ReflectComponent {
                 entity.insert(component);
             },
             apply: |mut entity, reflected_component| {
-                let mut component = entity.get_mut::<C>().unwrap();
+                if !C::Mutability::MUTABLE {
+                    panic!("This component is immutable, you cannot modify it through reflection");
```

Comment:
Nit: For debuggability, could we maybe have these panic messages indicate the method that caused the error? Even better would be to provide the name of the component.

Maybe something like: ``{component_name} is immutable, it cannot be modified via `ReflectComponent::apply` ``?

Reply:
I agree. I've updated the messages to call out the component (using `ShortName` for legibility) and the specific method that failed. I still view these as stop-gap measures, as we discussed earlier a `Result`-based fallible rewrite is the long-term solution here.